web/packages/teleterm/src/ui/DocumentDesktopSession/DocumentDesktopSession.tsx

```diff
@@ -83,8 +86,38 @@ export function DocumentDesktopSession(props: {
       }, new BrowserFileSystem())
   );
 
-  return (
-    <Document visible={props.visible}>
+  useListener(
+    client.onTransportOpen,
+    useCallback(
+      () => documentsService.update(uri, { status: 'connected' }),
+      [documentsService, uri]
+    )
+  );
+  useListener(
+    client.onTransportClose,
+    useCallback(
+      error => documentsService.update(uri, { status: error ? 'error' : '' }),
+      [documentsService, uri]
+    )
+  );
+  useListener(
+    client.onError,
+    useCallback(
+      () => documentsService.update(uri, { status: 'error' }),
+      [documentsService, uri]
+    )
+  );
```

Comment:
I should have combined these events while I was refactoring this, but well, I will keep it as is.